modules/Task/components/EditTaskDetails/EditTaskDetails.tsx

```diff
@@ -0,0 +1,117 @@
+import { Button, Modal, Select, TextInput, Textarea } from "@mantine/core";
+import { DateInput } from "@mantine/dates";
+import { useForm, SubmitHandler, Controller } from "react-hook-form";
+import "@mantine/dates/styles.css";
+
+import { TTask } from "@/shared/typedefs/types";
+import { EPriority } from "@/shared/typedefs/enums";
+import { useTasksContext } from "@/shared/utils/TasksProvider/TasksProvider";
+
+import { TEditTaskDetailsProps } from "./EditTaskDetails.types";
+
+function EditTaskDetails({
+  opened,
+  close,
+  task,
+  setTaskToEdit,
+}: TEditTaskDetailsProps) {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isValid },
+  } = useForm<TTask>({ mode: "onBlur" });
+
+  const { updateTask } = useTasksContext();
+
+  const onSubmit: SubmitHandler<TTask> = (data: TTask) => {
+    setTaskToEdit(null);
+    updateTask(data, task.id.toString());
+  };
+
+  const validateDueDate = (value: Date | null) => {
+    const currentDate = new Date();
+
+    if (value && value < currentDate) {
+      return "Due date cannot be in the past";
+    }
+
+    return true;
+  };
+
+  return (
+    <>
+      <Modal opened={opened} onClose={close} title="Edit Task" centered>
```

Comment:
Looks like you have declared the same modal in 3 places (CreateTask, EditTask and EditTaskDetails component). Please created a separate ticket to create a reusable modal (suggested name: `TaskModal`) and reuse in these 3 places instead of declaring in every use case.

Reply:
Also I don't see a valid use case for this new `EditTaskDetails` component, you can just trigger your existing`EditTask` component in `TaskDetails`. This new component is redundant.